src/k8s/pkg/component/storage.go

```diff
@@ -57,3 +67,36 @@ func DisableStorageComponent(s snap.Snap) error {
 
 	return nil
 }
+
+func ReconcileLocalStorageComponent(ctx context.Context, s snap.Snap, alreadyEnabled *bool, requestEnabled *bool, clusterConfig types.ClusterConfig) error {
+	var setDefault bool
+
+	if clusterConfig.LocalStorage.SetDefault != nil {
+		setDefault = *clusterConfig.LocalStorage.SetDefault
+	}
+
+	if vals.Deref(requestEnabled) && vals.Deref(alreadyEnabled) {
+		// If already enabled, and request does not contain `enabled` key
+		// or if already enabled and request contains `enabled=true`
+		err := UpdateStorageComponent(ctx, s, true, clusterConfig.LocalStorage.LocalPath, clusterConfig.LocalStorage.ReclaimPolicy, setDefault)
+		if err != nil {
+			return fmt.Errorf("failed to refresh local-storage: %w", err)
+		}
+		return nil
+	} else if vals.Deref(requestEnabled) {
+		err := UpdateStorageComponent(ctx, s, false, clusterConfig.LocalStorage.LocalPath, clusterConfig.LocalStorage.ReclaimPolicy, setDefault)
+		if err != nil {
+			return fmt.Errorf("failed to enable local-storage: %w", err)
+		}
+		return nil
+	} else if !vals.Deref(requestEnabled) {
+		err := DisableStorageComponent(s)
+		if err != nil {
+			return fmt.Errorf("failed to disable local-storage: %w", err)
+		}
+		return nil
+	} else {
+		// do nothing
+	}
```

Comment:
This is checking `if A { } else if !A { } else`. the final else is unnecessary.

Also in other places

```suggestion
	} else if vals.Deref(requestEnabled) {
		err := UpdateStorageComponent(ctx, s, false, clusterConfig.LocalStorage.LocalPath, clusterConfig.LocalStorage.ReclaimPolicy, setDefault)
		if err != nil {
			return fmt.Errorf("failed to enable local-storage: %w", err)
		}
		return nil
	} else {
		err := DisableStorageComponent(s)
		if err != nil {
			return fmt.Errorf("failed to disable local-storage: %w", err)
		}
		return nil
	}
```